docs/source/api/references/graphstorm.inference.rst

```diff
@@ -3,10 +3,11 @@
 graphstorm.inference
 ====================
 
-    GraphStorm inferrers assemble the distributed inference pipeline for different tasks.
+    GraphStorm Inferrers assemble the distributed inference pipelines for different tasks,
+    e.g., node classification, and link prediction.
 
-    If possible, users should always use these inferrers to avoid handling the distributed
-    inference processing.
+    If possible, users should always use these Inferrers to avoid handling the complexities
+    involved with the distributed data processing and model inference.
```

Comment:
Inference classes or inference objects would sound better, inferrer doesn't really exist as a word in the literature, and inferencer is also quite obscure.

We can keep it in the code, but change it for the docs?

Reply:
You can say GraphStorm inference module.

Reply:
`Inferrer` is a valid word, but not popular in literature. I updated it to `GraphStorm inference modules` and `GraphStorm training modules` accordingly to avoid ambiguity.